python/lsst/analysis/tools/atools/diffMatched.py

```diff
@@ -216,58 +206,53 @@ def finalize(self):
+            if self.name_prefix is None:
+                subtype = "chi" if self.compute_chi else "diff"
+                self.name_prefix = f"photom_mag_{{key_flux}}_{{name_class}}_{subtype}_"
+            if not self.produce.metric.units:
+                self.produce.metric.units = self.configureMetrics()
 
 
-# The diamond inheritance on MatchedRefCoaddTool seems ok
-class MatchedRefCoaddDiffMagMetric(MatchedRefCoaddDiffMagTool, MatchedRefCoaddDiffMetric):
-    """Metric for diffs between reference and measured coadd mags."""
-
-    def finalize(self):
-        super().finalize()
-        if self.unit is None:
-            self.unit = "" if self.compute_chi else "mmag"
-        if self.name_prefix is None:
-            subtype = "chi" if self.compute_chi else "diff"
-            self.name_prefix = f"photom_mag_{{key_flux}}_{{name_class}}_{subtype}_"
-        if not self.produce.metric.units:
-            self.produce.metric.units = self.configureMetrics()
-
-
-class MatchedRefCoaddDiffMagPlot(MatchedRefCoaddDiffMagTool, MagnitudeScatterPlot):
-    def finalize(self):
-        # TODO: Check if this is really necessary
-        # finalizing in this order should get all fluxes finalized before
-        # the MagnitudeScatterPlot looks for a flux to compute S/N from
-        MatchedRefCoaddDiffMagTool.finalize(self)
-        MagnitudeScatterPlot.finalize(self)
-        if not self.produce.yAxisLabel:
-            config = self.fluxes[self.mag_y]
-            label = f"{config.name_flux} - {self.fluxes['ref_matched'].name_flux}"
-            self.produce.yAxisLabel = f"chi = ({label})/error" if self.compute_chi else f"{label} (mmag)"
-
-
-class MatchedRefCoaddDiffPositionTool(MatchedRefCoaddDiffTool):
+class MatchedRefCoaddDiffPositionTool(MatchedRefCoaddDiffTool, MagnitudeScatterPlot):
     """Tool for diffs between reference and measured coadd astrometry."""
 
+    mag_sn = Field[str](default="cmodel_err", doc="Flux (magnitude) field to use for S/N binning on plot")
     scale_factor = Field[float](
         doc="The factor to multiply positions by (i.e. the pixel scale if coordinates have pixel units)",
         default=200,
     )
-    variable = ChoiceField[str](
-        doc="The astrometric variable to compute metrics for",
-        allowed={
-            "x": "x",
-            "y": "y",
-        },
+    coord_label = Field[str](
+        doc="The plot label for the astrometric variable (default coord_meas)",
+        optional=True,
+        default=None,
+    )
+    coord_meas = Field[str](
+        doc="The key for measured values of the astrometric variable",
+        optional=False,
+    )
+    coord_ref = Field[str](
+        doc="The key for reference values of the astrometric variabler",
         optional=False,
     )
 
     def finalize(self):
         # Check if it has already been finalized
         if not hasattr(self.process.buildActions, "diff"):
+            # Set before MagnitudeScatterPlot.finalize to undo PSF default
```

Comment:
Missing period at the end of this comment, unless you intend for these two lines to be read as one sentence (I think not).